src/deltalake_ext.rs

```diff
@@ -321,6 +342,39 @@ impl DeltaWriter {
 
         Ok(data_path)
     }
+
+    fn divide_by_partition_values(
+        &self,
+        values: Vec<Value>,
+    ) -> Result<HashMap<String, Vec<Value>>, DeltaWriterError> {
+        let mut partition_values: HashMap<String, Vec<Value>> = HashMap::new();
+
+        for value in values {
+            let key = self.json_to_partition_keys(&value)?;
+            match partition_values.get_mut(&key) {
+                Some(vec) => vec.push(value),
+                None => {
+                    partition_values.insert(key, vec![value]);
+                }
+            };
+        }
+
+        Ok(partition_values)
+    }
+
+    fn json_to_partition_keys(&self, value: &Value) -> Result<String, DeltaWriterError> {
+        if let Some(obj) = value.as_object() {
+            let key: Vec<String> = self
+                .partition_columns
+                .iter()
+                .map(|c| obj.get(c).unwrap_or(&Value::Null).to_string())
+                .collect();
+            return Ok(key.join("/"));
+        }
+
+        // todo change!
+        Err(DeltaWriterError::EmptyRecordBatch)
```

Comment:
is error message is not accurate right? it's an empty/invalid record, not record batch?

Reply:
yeah, you're right, was a leftover